packages/loader/container-loader/src/container.ts

```diff
@@ -627,13 +657,6 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         assert(this.loaded, "not loaded");
         assert(!this.closed, "closed");
 
-        // LoaderHeader.reconnect when set to false means we are allowing one connection,
-        // but do not allow re-connections. This is not very meaningful for attach process,
-        // plus this._canReconnect is provided to DeltaManager in constructor, so it's a bit too late.
-        // It might be useful to have an option to never connect, i.e. create file and close container,
-        // but that's a new feature to implement, not clear if we want to use same property for that.
-        assert(!(request.headers?.[LoaderHeader.reconnect] === false), "reconnect");
```

Comment:
not sure why it's removed. Should it stay and move with the rest of logic whenever next steps are taken?

Reply:
The point of this PR is that the `Container` should not be cracking open the `LoaderHeader`s of any `IRequest`, since that should be considered a layer violation.

As a result we get a very consistent/predictable result -- `Container` never does anything with respect to any `LoaderHeader` ever with this PR -- after all why would it, since it's not a `Loader` :)

If people are unhappy with the inability to configure reconnect behavior in the detached container flow, it seems one option would be for `Container.createDetached` to be augmented to take a config arg similar to `Container.load`.